broker-core/src/main/java/io/zeebe/broker/workflow/processor/SideEffectQueue.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.workflow.processor;
+
+import io.zeebe.broker.logstreams.processor.SideEffectProducer;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class SideEffectQueue implements SideEffectProducer, Consumer<SideEffectProducer> {
```

Comment:
This queue needs to allocate new objects every time. We should try to avoid object allocation in the "hot path".

Reply:
You mean by adding elements to the list we allocate new objects? The list itself is pre-allocated (should be marked final, my bad), and it is allocated once during the lifetime of a BpmnStepContext instance (which I think is allocated once during the lifetime of a BpmnStepProcessor instance).

What do you propose? Using ArrayList means we might have to resize the array anyway, but that might perform better if we always ensure some minimal size (e.g. 4).

Reply:
The `ArrayList` is ok. But I would not use the iterator because it is created every time. 

What do you think about changing the name of `accept()`? I think it is not clear that I can add multiple side effects.

Reply:
>What do you think about changing the name of accept()? I think it is not clear that I can add multiple side effects.

Vote for renaming